interview-questions.md

```diff
@@ -7,64 +7,64 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Rails follows an MVC pattern. What does that mean?
 
-  Your answer:
+  Your answer: MVC is model view controller which has CRUD actions.
 
-  Researched answer:
+  Researched answer:It basically works as following: Requests first come to the controller, controller finds an appropriate view and interacts with model which in turn interacts with database and send response to controller.
 
 
 
 2. What does it mean if an application has CRUD? What does CRUD stand for?
 
-  Your answer:
+  Your answer:CRUD means Create, Reate, Update, Destroy or Delete. it is the proccess for db's
 
-  Researched answer:
+  Researched answer: CRUD is an acronym that comes from the world of computer programming and refers to the four functions that are considered necessary to implement a persistent storage application: create, read, update and delete.
 
 
 
 3. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: relational are databases using a model and schema
 
-  Researched answer:
+  Researched answer: Relational databases are also called Relational Database Management Systems (RDBMS) or SQL databases. A relational database is a digital database based on the relational model of data, as proposed by E. F. Codd in 1970. A software system used to maintain relational databases is a relational database management system.
 
 
 
 4. What are the 5 HTTP verbs? What are they important?
 
-  Your answer:
+  Your answer: GET POST PUT PATCH UPDATE DELETE
 
-  Researched answer:
+  Researched answer:The valid HTTP methods are GET , HEAD , POST , PUT , DELETE , TRACE , OPTIONS , CONNECT , and PATCH .
 
 
 
 5. What is object-relational mapping?
 
-  Your answer:
+  Your answer: is the ability to write queries
 
-  Researched answer:
+  Researched answer: Object-relational mapping in computer science is a programming technique for converting data between incompatible type systems using object-oriented programming languages.
 
 
 
 6. What is a gem?
 
-  Your answer:
+  Your answer: Gems are package managers for Ruby
```

Comment:
Yup, RubyGems is the package manager while the gems themselves is pre-written code you can use in a Ruby project